Data/Graph/Inductive/PatriciaTree.hs

```diff
@@ -220,6 +224,25 @@ toAdj = concatMap expand . IM.toList
 fromAdj :: Adj b -> IntMap [b]
 fromAdj = IM.fromListWith addLists . map (second (:[]) . swap)
 
+data FromListCounting a = FromListCounting !Int !(IntMap a)
```

Comment:
Yes, it's solely for `!`. I can add instances, I suppose, but no exposed function ever returns anything of that type, so I don't really see the point of generating that code.